cli/report.go

```diff
@@ -109,28 +115,44 @@ type testEvent struct {
 }
 
 type reporter struct {
-	input         <-chan *testEvent
-	client        *gorp.Client
-	launchName    string
-	launchID      string
-	launchOnce    sync.Once
-	tests         map[string]string
-	suites        map[string]string
-	logs          []*gorp.SaveLogRQ
-	logsBatchSize int
-	waitQueue     sync.WaitGroup
+	input            <-chan *testEvent
+	client           *gorp.Client
+	launchName       string
+	launchID         string
+	launchOnce       sync.Once
+	launchAttributes []*gorp.Attribute
+	tests            map[string]string
+	suites           map[string]string
+	logs             []*gorp.SaveLogRQ
+	logsBatchSize    int
+	waitQueue        sync.WaitGroup
 }
 
-func newReporter(client *gorp.Client, launchName string, input <-chan *testEvent) *reporter {
+func newReporter(client *gorp.Client, launchName string, input <-chan *testEvent, launchAttrArgs ...string) *reporter {
+	launchAttributes := make([]*gorp.Attribute, 0, len(launchAttrArgs))
+	for _, attr := range launchAttrArgs {
+		key, value, valid := strings.Cut(attr, ":")
+		if valid {
```

Comment:
i would suggest that we still use key with empty value in case there is no separator passed (e.g. what reportportal previously called "tags")

Reply:
Submitting an empty value results in the service API returning an error:
```
{"errorCode":4001,"message":"Incorrect Request. [Field 'attributes[].value' should not contain only white spaces and shouldn't be empty.] "}
```

Reply:
@mjte-riot yeah, sorry for misleading instructions. If there's no separator (e.g. old 'tag'), the whole value goes to `value` field, not to the `key` field

Reply:
No problem! That's updated now